src/app-layout/index.tsx

```diff
@@ -73,10 +76,27 @@ const AppLayout = React.forwardRef(
     const props = { contentType, headerSelector, footerSelector, ...rest, ariaLabels };
 
     const baseProps = getBaseProps(rest);
+    const ownDrawers = (props as any).drawers;
+    const disableRuntimeDrawers = (props as any).__disableRuntimeDrawers;
+    const combinedDrawers = [...runtimeDrawers.before, ...(ownDrawers?.items ?? []), ...runtimeDrawers.after];
+    const finalDrawers = combinedDrawers.length > 0 ? { ...ownDrawers, items: combinedDrawers } : ownDrawers;
+
+    useEffect(() => {
+      if (disableRuntimeDrawers) {
+        return;
+      }
+      return awsuiPluginsInternal.appLayout.registerAppLayout(drawers =>
+        setRuntimeDrawers(convertRuntimeDrawers(drawers))
```

Comment:
Thinking about re-renders in this sort of scenario:

- A custom drawer is open
- Something else on the page registers a new custom drawer
- Result: I think at this point the visible custom drawer would re-render? I'm not _sure_ that I would expect this as a drawer developer

wdyt?

Reply:
This line in particular does not cause the drawers content to be remounted. If keys management in JSX is done correctly, it will work fine.

I will keep an eye on this use-case, but after this PR